remote/remote.go

```diff
@@ -53,7 +58,34 @@ func WithPreviousImage(imageName string) ImageOption {
 func FromBaseImage(imageName string) ImageOption {
 	return func(r *Image) (*Image, error) {
 		var err error
-		r.image, err = newV1Image(r.keychain, imageName)
+
+		currentImagePlatform, err := imagePlatform(r.image)
```

Comment:
It looks like `WithDefaultPlatform` option must be passed before the `FromBaseImage` option to get the correct value here. That might not be obvious to library consumers. I want to change `FromBaseImage` to take an image rather than a string which would make this case easier, but in the meantime we probably need to solve the ordering problem.

We could potentially change the options to produce a config that is then applied or instead of mutating the image direction in option func.

Reply:
Thanks for the feedback. I added `WithDefaultPlatform` without a clear use-case yet, except to exercise the some of the `newV1Image` fallback logic that is now removed. Would it be ok to remove `WithDefaultPlatform` for now, the we can add it back later if there's a need that 'FromBaseImage` doesn't provide?

Reply:
I think that makes sense. Generating a valid empty windows image isn't really possible anyways it seems like (you need a base layer with a lot of configuration).